### PR TITLE
Simplify the OpInfos for norm / linalg_norm

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76551
* #76547
* __->__ #76545

This will be helpful later on when we add tests for `dtype` (after correcting its gradient formula).

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1842,6 +1842,9 @@ def sample_inputs_norm(op_info, device, dtype, requires_grad, **kwargs):
         ((S, S), (-2,), 'neg_2'),
         ((S, S), (-0.5,), 'neg_0_5'),
         ((S, S), (-1.5,), 'neg_1_5'),
+        # FIXME gradgrad and noncotiguous_samples fails for these on CPU, how about that
+        # ((S, S), (inf,), 'inf'),
+        # ((S, S), (-inf,), 'neg_inf'),
```
Comment (Lezcano) [inline torch/testing/_internal/common_methods_invocations.py:1847]: This was an interesting find, and it seems to be related to the odd fix `linalg.vector_norm` has. See https://github.com/pytorch/pytorch/pull/76547#discussion_r861201122.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1932,63 +1930,14 @@ def sample_inputs_norm_inf(op_info, device, dtype, requires_grad, **kwargs):
...
-                args=(ord,),
-                kwargs=dict(
-                    keepdim=keepdim,
-                    dim=dim)))
+    sizes = ((S,), (2, 2))
+    dims = (None, 0, -1)
+    ords = (inf, 2, 1, 0, 0.9, -2.1, -inf)
 
-    return inputs
+    for size, ord_, dim, keepdim in product(sizes, ords, dims, (True, False)):
```
Comment (ngimel) [inline torch/testing/_internal/common_methods_invocations.py:1939]: you are no longer testing tuple of dims for 2d inputs?

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:1855]: You can conditionally include these when CUDA samples are requested

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (Lezcano) [reply torch/testing/_internal/common_methods_invocations.py:1939]: I had that in mind, and then I forgot to add it. Thank you for the catch!

**Comment by Lezcano** [comment]
This is ready for another review and perhaps a merge @ngimel @mruberry

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -1910,7 +1910,13 @@ def sample_inputs_norm(op_info, device, dtype, requires_grad, **kwargs):
         ((S, S), (-2,), 'neg_2'),
         ((S, S), (-0.5,), 'neg_0_5'),
         ((S, S), (-1.5,), 'neg_1_5'),
-    )
+    ]
+
+    # FIXME gradgrad and noncotiguous_samples fail on inf and -inf norms on CPU because of vectorization
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1915]: Nice note and device type check

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -1937,15 +1943,10 @@ def sample_inputs_norm(op_info, device, dtype, requires_grad, **kwargs):
...
     )
 
-    cases_negdim = []
-    for case in cases_negdim_base:
-        cases_negdim.append(case)
-        shape, args, name = case
-        new_args = copy.deepcopy(list(args))
-        new_args[1] *= -1
-        cases_negdim.append((shape, tuple(new_args), name.replace("_dim", "_neg_dim")))
+    cases_negdim = ((shape, args[:1] + (-args[1],) + args[2:], name.replace("_dim", "_neg_dim"))
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1946]: Comment here please

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -1992,63 +1993,19 @@ def sample_inputs_norm_inf(op_info, device, dtype, requires_grad, **kwargs):
...
-                make_tensor(
-                    test_size, dtype=dtype, device=device,
-                    low=None, high=None,
-                    requires_grad=requires_grad),
-                args=(ord,),
-                kwargs=dict(
-                    keepdim=keepdim,
-                    dim=dim)))
+    for size, ord_, keepdim in product(sizes, ords, (True, False)):
+        for dim in dims:
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2003]: Why a separate iteration for dims here?

Comment (Lezcano) [reply torch/testing/_internal/common_methods_invocations.py:2003]: To be able to add the extra dim case of `(-1, 0)` when `len(size) >= 2`.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:2003]: Oh, right, sorry it just appeared so separated on GitHub I failed to make the connection

**Review by mruberry — APPROVED** [review]
Cool! LGTM
